packages/ui/src/components/NavigationMenu/api.ts

```diff
@@ -0,0 +1,28 @@
+import { InjectionKey, inject, Ref } from 'vue'
+import { forceInject } from '@/util'
+
+export interface MenuApi {
+  /**
+   * Should be called inside item to select it
+   */
+  select: (index: string) => void
+  active: Ref<string>
+  collapsed: Ref<boolean>
+}
+
+export interface SubmenuApi {
+  /**
+   * Should be called inside item when created
+   */
+  register: (index: Ref<string>) => void
+  /**
+   * Should be called inside item when destroyed
+   */
+  unregister: (index: Ref<string>) => void
+}
```

Comment:
I think its better to move `unregister` logic inside of `register` (1) without need to pass the same `Ref` twice and (2) preventing caller from forgetting to unregister on scope dispose.

Reply:
Also it would be nice to mention that `register` should be called during `setup` and that it will be unregistered automatically on scope dispose.